marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -103,126 +90,70 @@ handleAction Init = do
   mCachedWalletJson <- liftEffect $ getItem walletLocalStorageKey
   for_ mCachedWalletJson \json ->
     for_ (runExcept $ decodeJSON json) \cachedWallet ->
-      assign _subState $ Right $ mkWalletState cachedWallet
+      assign _subState $ Right $ Play.mkInitialState cachedWallet
   -- TODO: fetch contract templates from the library ??
   assign _templates templates
 
--- pickup actions
-handleAction (SetPickupCard pickupCard) = assign (_pickupState <<< _card) pickupCard
+handleAction (SetNewWalletNickname nickname) = assign (_newWalletNicknameKey <<< _nickname) nickname
+
+handleAction AddNewWallet = do
+  oldWallets <- use _wallets
+  newWalletNicknameKey <- use _newWalletNicknameKey
+  when (not $ member newWalletNicknameKey oldWallets) do
+    modify_
```

Comment:
this way of grouping seems a little weird... I'd prefer to use something like the following (but totally optional)

```
modify_ (
  over _wallets (insert newWalletNicknameKey defaultWalletDetails)
  <<< set _newWalletNicknameKey mempty
  <<< set (_playState <<< _card) mempty
)
```

Reply:
I agree, that is the style that is used elsewhere

Reply:
Agreed.